include/rx/ranges.hpp

```diff
@@ -1998,6 +2001,58 @@ struct reverse {
     }
 };
 
+/// Copy values of a range into a container during iteration.
+template <class Dest>
+struct tee {
+    Dest& dest;
+    explicit constexpr tee(Dest& dest) noexcept : dest(dest) {}
+
+    template <class R>
+    struct Range {
+        R input;
+        Dest& dest;
+        mutable bool sunk = false;
+
+        using output_type = get_output_type_of_t<R>;
+        static constexpr bool is_finite = is_finite_v<R>;
+        static constexpr bool is_idempotent = false;
+
+        template <class Rx>
+        constexpr Range(Rx&& input, Dest& dest) noexcept : input(std::forward<Rx>(input)), dest(dest) {}
+
+        constexpr void next() {
+            if (!sunk) {
+                sink_one(input.get(), dest);
+            }
+            input.next();
+            sunk = false;
+        }
+
+        constexpr output_type get() const {
+            auto&& value = input.get();
+            sink_one(value, dest);
```

Comment:
Why sink during `get()`?

It seems like sinking during `next()` should be good enough, and do away with the `mutable bool sunk` member.